From 1eae2af78f76a836ab4cb23f7bddd28c4432f24a Mon Sep 17 00:00:00 2001
From: Joe Baldino <pedanticdm@gmx.us>
Date: Mon, 3 Apr 2023 22:12:19 -0700
Subject: [PATCH] ARCHLINUX: conform Makefile to packaging standards

LDFLAGS are not used by the upstream build system. Fix that to allow makepkg to
do its work.

Remove "Dad's CFLAGS". Only keep essential arguments to build source code, and
leave additional optimizations, etc to the end-user.
---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 2f6e160d53b1..bbf30a443c7c 100644
--- a/Makefile
+++ b/Makefile
@@ -10,10 +10,10 @@ OBJ=$(SRC:$(SRCDIR)/%.c=$(BUILDDIR)/%.o)
 DEP=$(SRC:$(SRCDIR)/%.c=$(BUILDDIR)/%.d)
 
 NCURSES_LIB?=ncurses
-NCURSES_CFLAGS?=`pkg-config --cflags $(NCURSES_LIB)`
-NCURSES_LDLIBS?=`pkg-config --libs $(NCURSES_LIB)`
+NCURSES_CFLAGS?=$(shell pkg-config --cflags $(NCURSES_LIB))
+NCURSES_LDLIBS?=$(shell pkg-config --libs $(NCURSES_LIB))
 
-CFLAGS?=-Wall -Wextra -pedantic -std=c11 -O2 -march=native -D_GNU_SOURCE $(NCURSES_CFLAGS)
+CFLAGS+= -Wall -Wextra -pedantic -std=c11 -D_GNU_SOURCE $(NCURSES_CFLAGS)
 LDLIBS?=$(NCURSES_LDLIBS)
 
 PREFIX?=/usr/local
@@ -25,7 +25,7 @@ BINDIR?=$(PREFIX)/bin
 all: $(TARGET)
 
 $(TARGET): $(OBJ)
-	$(CC) $(CFLAGS) $^ -o $(TARGET) $(LDLIBS)
+	$(CC) $(CFLAGS) $^ -o $(TARGET) $(LDFLAGS) $(LDLIBS)
 
 %.o : %.c
 
-- 
2.40.0

